src/main/scala/resource/package.scala

```diff
@@ -77,6 +77,56 @@ package object resource {
     toReturn
   }
 
+  /**
+    * Creates a [[ManagedResource]] shared by all users for any type with a [[Resource]] type class implementation.
+    *
+    * There is only one instance of the resource at the same time for all the users.
+    * The instance will be closed once no user is still using it.
+    */
+  def shared[A: Resource : Manifest](opener: => A): ManagedResource[A] = {
+    @volatile var sharedReference: Option[(Int, A)] = None
+    val lock = new AnyRef
+
+    def acquire = {
+      lock.synchronized {
```

Comment:
I like the idea od this, but I'd like to try a lockless strategy here.  Would you mind if I add that after merging?